File: Data Structures & Algorithms/longest-palindromic-substring/submission-0.py
class Solution:
    def longestPalindrome(self, s: str) -> str:
        long_substr = ""

        for i in range(len(s)):
            for j in range(i, len(s)):
                substr = s[i:j+1]
                l, r = 0, len(substr) - 1

                while l <= r:
                    if substr[l] != substr[r]:
                        break

                    l, r = l + 1, r - 1
                
                if l > r and len(substr) > len(long_substr):
                    long_substr = substr

        return long_substr

                    

File: Data Structures & Algorithms/word-break/submission-0.cpp
class Solution {
public:
    bool wordBreak(string s, vector<string>& wordDict) {
        int n = s.size();
        vector<bool> dp(n + 1, false);
        dp[n] = true;
        for (int i = n - 1; i > -1; i--) {
            for (auto w : wordDict) {
                if (i + w.size() <= n) {
                    string cur = s.substr(i, w.size());
                    if (cur == w) {
                        dp[i] = dp[i + w.size()];
                    }
                }
                if (dp[i]) break;
            }
        }

        return dp[0];
    }
};
